File: aoc2015/day1/part2.js
const fs = require('fs').promises;


async function main() {
  const file = await fs.readFile('./input.txt', 'utf8');
  const fileContents = file.trim().split('');


  let curr = 0;
  let posn = 0;

  for (let i = 0; i < fileContents.length; i++) {
    if (fileContents[i] === '(') {
      curr += 1;
      posn += 1;
    }
    else if (fileContents[i] === ')') {
      curr -= 1;
      posn += 1;
      if (curr == -1) {
        console.log(posn);
        break;
      }
    }

  }
}



try {
  main();
}
catch (err) {
  console.log(err);
}

File: aoc2018/src/day1/part2.go
package main

import (
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func freqCount(input string) int {
	var sum int
	seen := map[int]bool{}
	for {
		for _, line := range strings.Split(input, "\n") {
			val, err := strconv.Atoi(line)
			if err != nil {
				log.Fatal(err)
			}
			sum += val
			if seen[sum] {
				return sum
			}
			seen[sum] = true
		}
	}
}

func main() {
	content, err := os.ReadFile("./input.txt")
	if err != nil {
		log.Fatal(err)
	}
	strcontent := string(content)
	fmt.Println(freqCount(strings.TrimRight(strcontent, "\n")))

}
